hypervisor/arch/x86/guest/vmcall.c

```diff
@@ -105,6 +106,24 @@ static const struct hc_dispatch hc_dispatch_table[] = {
 		.permission_flags = GUEST_FLAG_SECURE_WORLD_ENABLED},
 };
 
+uint16_t allocate_dynamical_vmid(struct acrn_vm_creation *cv)
+{
+	uint16_t vm_id;
+	struct acrn_vm_config *vm_config;
+
+	spinlock_obtain(&vm_id_lock);
+	vm_id = get_unused_vmid();
+	if (vm_id != ACRN_INVALID_VMID) {
+		vm_config = get_vm_config(vm_id);
+		memcpy_s(vm_config->name, MAX_VM_NAME_LEN, cv->name, MAX_VM_NAME_LEN);
```

Comment:
Where do you enforce there are no space characters in these VM names (matching the  "\S+" pattern restriction in the .xsd file)

Reply:
here we have a assumption that the vm name passed from devicemodel has no space characters. From HV p.o.v, it is just need a string comparison and show in shell, it will not care if any space or special char in it.